harper-core/src/mask/mod.rs

```diff
@@ -8,7 +8,7 @@ use crate::Span;
 /// This is primarily used by [`crate::parsers::Mask`] to create parsers for
 /// things like comments of programming languages.
 pub trait Masker: Send + Sync {
-    fn create_mask(&mut self, source: &[char]) -> Mask;
+    fn create_mask(&self, source: &[char]) -> Mask;
```

Comment:
Why remove mutability here? Not against it, just want to know your reasoning.

Reply:
Because of this, `parse()` needs `&mut self`, and because of that, every parser needs to be a mutable reference, so it cannot be cloned or copied, even though they are almost all unit structs. This creates some issues using them in closures, where ownership is passed to the closure.